File: system-test/dotnet/Dsl.Core/UseCase/UseCases/Base/BaseMyShopUseCase.cs
using Driver.Port;
using Dsl.Core.Shared;
using Driver.Port.Dtos.Error;

namespace Dsl.Core.UseCase.UseCases.Base;

public abstract class BaseMyShopUseCase<TResponse, TVerification>
    where TVerification : ResponseVerification<TResponse>
{
    protected readonly IMyShopDriver _driver;
    protected readonly UseCaseContext _context;

    protected BaseMyShopUseCase(IMyShopDriver driver, UseCaseContext context)
    {
        _driver = driver;
        _context = context;
    }

    public abstract Task<MyShopUseCaseResult<TResponse, TVerification>> Execute();
}

File: system/multitier/backend-java/src/testSupport/java/com/mycompany/myshop/backend/support/core/usecase/usecases/base/BaseMyShopUseCase.java
package com.mycompany.myshop.backend.support.core.usecase.usecases.base;

import com.mycompany.myshop.backend.support.BackendDriver;
import com.mycompany.myshop.backend.support.core.shared.BaseUseCase;
import com.mycompany.myshop.backend.support.core.shared.ResponseVerification;

public abstract class BaseMyShopUseCase<R, V extends ResponseVerification<R>>
        extends BaseUseCase<BackendDriver, R, V> {

    protected BaseMyShopUseCase(BackendDriver driver) {
        super(driver);
    }
}
